### DISCUSSION TITLE
Admins collection > User collection

### DISCUSSION THREAD

**Discussion by jakehopking** [payloadcms/payload#1639] [General]
I had my previous access setup very basic, and I want to move to a more granular setup which includes roles, with field level access and to integrated with drafts etc.

I've tried copying the way you do so from the `website-cms` repo, however I'm now locked out of my admin panel. The login dialog won't render :/ which suggests I've missed something fundamental. Surely the dialog should render even if I change the access collection?

Any guidance here? Happy to share any code you need...

### My previous Admins collection was like this:
```tsx
import { CollectionConfig } from 'payload/types';

export const Admins: CollectionConfig = {
  slug: 'admins',
  admin: {
    useAsTitle: 'title',
  },
  auth: {
    tokenExpiration: 7200, // How many seconds to keep the user logged in
    verify: true, // Require email verification before being allowed to authenticate
    maxLoginAttempts: 5, // Automatically lock a user out after X amount of failed logins
    lockTime: 600 * 1000, // Time period to allow the max login attempts
    // More options are available
  },
  fields: [
    {
      name: 'firstName',
      label: 'First Name',
      type: 'text',
    },
    {
      name: 'lastName',
      label: 'Last Name',
      type: 'text',
    },
    {
      name: 'email',
      label: 'Email',
      type: 'email',
    },
    {
      name: 'role',
      type: 'select',
      required: true,
      defaultValue: 'admin',
      options: ['user', 'admin', 'editor', 'developer'],
    },
  ],
};

```

### And I've moved to your implementation of a User collection.

```tsx
import { CollectionConfig } from 'payload/types';
import { MediaType } from '../tokens/images';
import { isAdmin, isAdminFieldLevel } from '../access/isAdmin';
import {
  isAdminOrSelf,
  isAdminOrSelfFieldLevel,
} from '../access/isAdminOrSelf';
import { enumToFieldOptions } from '../utilities';

export enum UserRole {
  admin = 'admin',
  public = 'public',
}

export interface UserType {
  id: string;
  firstName: string;
  lastName: string;
  twitter?: string;
  // photo?: string | MediaType;
  roles: UserRole[];
  email?: string;
  resetPasswordToken?: string;
  resetPasswordExpiration?: string;
  loginAttempts?: number;
  lockUntil?: string;
  createdAt: string;
  updatedAt: string;
}

export const User: CollectionConfig = {
  slug: 'users',
  auth: true,
  admin: {
    useAsTitle: 'email',
  },
  access: {
    create: isAdmin,
    read: () => true,
    update: isAdminOrSelf,
    delete: isAdminOrSelf,
  },
  fields: [
    {
      type: 'row',
      fields: [
        {
          name: 'firstName',
          type: 'text',
          required: true,
        },
        {
          name: 'lastName',
          type: 'text',
          required: true,
        },
      ],
    },
    {
      name: 'twitter',
      type: 'text',
    },
    // {
    //   name: 'photo',
    //   type: 'upload',
    //   relationTo: 'media',
    // },
    {
      name: 'roles',
      type: 'select',
      hasMany: true,
      defaultValue: [UserRole.public],
      required: true,
      access: {
        read: isAdminOrSelfFieldLevel,
        create: isAdminFieldLevel,
        update: isAdminFieldLevel,
      },
      options: enumToFieldOptions(UserRole),
    },
  ],
};

```

**Comment by jakehopking** [comment]
The user collection has been created in mongodb, and I can see that drafts has been enabled too. 
This suggested that the code doesn't have errors as it has been parsed?

Do I need to create and admin user document on mongodb to enable login? If so, how can I do this?

**Reply by jakehopking** [reply]
Pages in my app are now returning undefined for slugs too, but are still there if I manually enter the url slug in the browser address bar. I presume this is because I now need to filter results with a `_status: 'published'`

However looking at existing docs in my pages collection in mongodb, I can't see this `_status` being applied.

**Comment by jmikrut** [comment]
If the admin panel isn't rendering at all, then I'd open your inspector, look at your Network tab, and try and see if there is a crashing request being made. Maybe one of your access control functions is crashing. I'd start there.

**Reply by jakehopking** [reply]
Thanks @jmikrut. I've pasted verbatim the code from your `website` repo and I can now see the login dialog. However I can't login (which is expected)... How can I reset passwords/create an admin user?

**Reply by jakehopking** [reply]
Not to worry.... I've removed my user from mongodb, and I can now see the user creation form 👍

**Reply by jmikrut** [reply]
Awesome! Good work!
